### ISSUE TITLE
Including disko to a custom, off-line installer

### ISSUE THREAD

**Issue by vilvo** [nix-community/disko#408]
I've tested including disko to a custom, nixos-based off-line installer. cli and lib can be included easily but I found that `make-disk-image.nix` gets evaluated in a way that requires network connection to download the dependencies. Networked works fine but off-line does not.

I think I could make the `diskoLib` internal dependency to `make-disk-image.nix` included in our installer image configuration but that does not sound right as future changes to this disko internal dependency may change.

Is it sensible for disko to support this kind of off-line installer?

**Comment by phaer** [comment]
> cli and lib can be included easily but I found that make-disk-image.nix gets evaluated in a way that requires network connection to download the dependencies. 

Are you intending to use make-disk-image.nix in your offline installer, i.e. create pre-built disk images there? Because make-disk-image.nix should not be required to just run disko from an installer if that hasn't been changed recently - so maybe evaluation is just a bit too eager?

**Comment by vilvo** [comment]
> > cli and lib can be included easily but I found that make-disk-image.nix gets evaluated in a way that requires network connection to download the dependencies.
> 
> Are you intending to use make-disk-image.nix in your offline installer, i.e. create pre-built disk images there? 

Yes. Basically include disko with deps and partitioning declaration in the offline installer which includes pre-built disk image(s).
